GVFS/GVFS.Notifications/VFSForGit.Mac/VFSForGit.UnitTests/ProductInfoFetcherTests.m

```diff
@@ -0,0 +1,111 @@
+#import <XCTest/XCTest.h>
+#import "ProductInfoFetcher.h"
+#import "MockNSTask.h"
+
+@interface ProductInfoFetcherTests : XCTestCase
+@end
+
+@implementation ProductInfoFetcherTests
+
+- (void)testProductInfoFetcherReturnsVFSForGitVersion
+{
+    NSString * const ExpectedVersion = @"GVFS Version 1.0.94768.54";
+    ProductInfoFetcher *productInfoFetcher = [self
+        infoFetcherWithExpectedOutput:ExpectedVersion
+        launchError:nil
+        cmdError: nil
+        exitCode:0];
+    NSString *vfsforGitVersion;
+    NSError *error;
+    
+    XCTAssertTrue([productInfoFetcher tryGetVFSForGitVersion:&vfsforGitVersion error:&error]);
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(vfsforGitVersion, ExpectedVersion);
+}
+
+- (void)testProductInfoFetcherReturnsGitVersion
+{
+    NSString * const ExpectedVersion = @"git version 2.20.1.vfs.1.1.104.g2ab73608a0";
+    ProductInfoFetcher *productInfoFetcher = [self
+        infoFetcherWithExpectedOutput:ExpectedVersion
+        launchError:nil
+        cmdError:nil
+        exitCode:0];
+    NSString *gitVersion;
+    NSError *error;
+    
+    XCTAssertTrue([productInfoFetcher tryGetGitVersion:&gitVersion error:&error]);
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(gitVersion, ExpectedVersion);
+}
+
+- (void)testProductInfoFetcherReturnsNilOnLaunchErrors
+{
+    NSString * const ErrorString = @"git command not found";
+    ProductInfoFetcher *productInfoFetcher = [self
+        infoFetcherWithExpectedOutput:nil
+        launchError:nil
```

Comment:
If we're testing launch errors in this test should `launchError` be non-nil?

Reply:
right. corrected the test.